sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,641 @@
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+
+public class HadoopInputFormatIO {
+
+	private static final Logger LOG = LoggerFactory.getLogger(HadoopInputFormatIO.class);
+
+	public static <K,V> Read<K,V> read() {
+		return new AutoValue_HadoopInputFormatIO_Read.Builder<K,V>().build();
+	}
+
+	@AutoValue
+	public abstract static class Read<K, V> extends PTransform<PBegin, PCollection<KV<K, V>>> {
+
+		private static final long serialVersionUID = 1L;
+		private TypeDescriptor<K> keyClass;
+		private TypeDescriptor<V> valueClass;
+		private Coder<K> keyCoder;
+		private Coder<V> valueCoder;
+
+		abstract Builder<K, V> toBuilder();
+		@Nullable public abstract  SimpleFunction< ?, K> getSimpleFuncForKeyTranslation();
+		@Nullable public abstract  SimpleFunction< ?, V> getSimpleFuncForValueTranslation();
+		@Nullable public abstract  SerializableConfiguration getConfiguration();
+
+		public void setKeyClass(TypeDescriptor<K> keyClass) {
+			this.keyClass = keyClass;
+		}
+
+		public void setValueClass(TypeDescriptor<V> valueClass) {
+			this.valueClass = valueClass;
+		}
+
+		public Coder<K> getKeyCoder() {
+			return keyCoder;
+		}
+
+		public Coder<V> getValueCoder() {
+			return valueCoder;
+		}
+
+		public Read<K, V> withConfiguration(Configuration configuration) {
+			checkNotNull(configuration,  "Configuration cannot be null.");
+			return toBuilder().setConfiguration(new SerializableConfiguration(configuration)).build();
+		}
+
+		public Read<K, V> withKeyTranslation(SimpleFunction<?, K> simpleFuncForKeyTranslation) {
+			checkNotNull(simpleFuncForKeyTranslation, "Simple function for key translation cannot be null.");
+			return toBuilder().setSimpleFuncForKeyTranslation(simpleFuncForKeyTranslation).build();
+		}
+
+		public Read<K, V> withValueTranslation(SimpleFunction< ?, V> simpleFuncForValueTranslation) {
+			checkNotNull(simpleFuncForValueTranslation, "Simple function for value translation cannot be null.");
+			return toBuilder().setSimpleFuncForValueTranslation(simpleFuncForValueTranslation).build();
+		}
+
+		@AutoValue.Builder
+		abstract static class Builder<K, V> {
+			abstract Builder<K,V> setConfiguration(SerializableConfiguration configuration);
+			abstract Builder<K,V> setSimpleFuncForKeyTranslation(SimpleFunction< ?,K> simpleFuncForKeyTranslation);
+			abstract Builder<K,V> setSimpleFuncForValueTranslation(SimpleFunction<?,V> simpleFuncForValueTranslation);
+			abstract Read<K,V> build();
+		}
+
+		public void validate(PBegin input) {
+			checkNotNull(this.getConfiguration(), "Need to set the configuration of a HadoopInputFormatIO Read using method Read.withConfiguration().");
+			String inputFormatClassProperty = this.getConfiguration().getConfiguration().get("mapreduce.job.inputformat.class") ;
+			if (inputFormatClassProperty == null) {
+				LOG.error("Hadoop InputFormat class property \"mapreduce.job.inputformat.class\" is not set in configuration.");
```

Comment:
Removed the exception logging, as we throw it. Had referred the BigTable IO, where this is done